torch/distributed/device_mesh.py

```diff
@@ -1124,30 +1141,13 @@ def _create_unflatten_mesh(
             # TODO: To make backend init more efficient with cute layout representation and support
             # per dim backend init.
             if hasattr(self, "_dim_group_names"):
-                unflatten_length = len(mesh_sizes)
-                unflatten_layout = _MeshLayout(
-                    tuple(unflattened_layout.sizes[dim : dim + unflatten_length]),  # type: ignore[index]
-                    tuple(unflattened_layout.strides[dim : dim + unflatten_length]),  # type: ignore[index]
+                dim_group_names = self._dim_group_names.copy()
+                dim_group_names[dim : dim + 1] = self._init_process_groups(
+                    partial_layout,
+                    root_mesh._global_rank_permutation,
+                    mesh_dim_names,
+                    backend_override,
```

Comment:
I like this one so that we can reuse it later if we only want to init backend for some dims not all. And also to make _flatten generic like what Tensor is doing also needs this.